StudentList/Fragments/StudentListFragment.cs

```diff
@@ -119,15 +118,18 @@ private void StudentAdapterItemClick(object sender, Student e)
 
         private async void Reset()
         {
-            this.loadingProgressBar.Visibility = ViewStates.Visible;
-            this.filteringResultTextView.Visibility = ViewStates.Invisible;
-
-            var students = await this.repository.GetStudentsAsync(StudentFilter.Default);
+            await this.Activity.RunMethodWithLoaderAsync(this.PutStudentsToAdapter(StudentFilter.Default));
             this.studentsCountTextView.Text = string.Format(
-                CultureInfo.InvariantCulture, this.GetString(Resource.String.student_count_pattern), students.Count);
+               CultureInfo.InvariantCulture, this.GetString(Resource.String.student_count_pattern), this.studentAdapter.ItemCount);
+        }
 
-            this.loadingProgressBar.Visibility = ViewStates.Invisible;
+        private async Task PutStudentsToAdapter(StudentFilter studentFilter)
+        {
+            var students = await this.repository.GetStudentsAsync(studentFilter);
```

Comment:
You wrapped wrong function with loader. You should've wrapped repo call, not call to PutStudentsToAdapter as PutStudentsToAdapter is a UI function, and repo call is async function.